src/hash.rs

```diff
@@ -0,0 +1,434 @@
+                $from([0; $size])
+            }
+
+            /// Get the size of this object in bytes.
+            pub fn len() -> usize {
+                $size
+            }
+
+            #[inline]
+            /// Assign self to be of the same value as a slice of bytes of length `len()`.
+            pub fn clone_from_slice(&mut self, src: &[u8]) -> usize {
+                let min = cmp::min($size, src.len());
+                self.0[..min].copy_from_slice(&src[..min]);
+                min
+            }
+
+            /// Convert a slice of bytes of length `len()` to an instance of this type.
+            pub fn from_slice(src: &[u8]) -> Self {
+                let mut r = Self::new();
+                r.clone_from_slice(src);
+                r
+            }
+
+            /// Copy the data of this object into some mutable slice of length `len()`.
+            pub fn copy_to(&self, dest: &mut[u8]) {
+                let min = cmp::min($size, dest.len());
+                dest[..min].copy_from_slice(&self.0[..min]);
+            }
+
+            /// Returns `true` if all bits set in `b` are also set in `self`.
+            pub fn contains<'a>(&'a self, b: &'a Self) -> bool {
+                &(b & self) == b
+            }
+
+            /// Returns `true` if no bits are set.
+            pub fn is_zero(&self) -> bool {
+                self.eq(&Self::new())
+            }
+
+            /// Returns the lowest 8 bytes interpreted as a BigEndian integer.
+            pub fn low_u64(&self) -> u64 {
+                let mut ret = 0u64;
+                for i in 0..min($size, 8) {
+                    ret |= (self.0[$size - 1 - i] as u64) << (i * 8);
+                }
+                ret
+            }
+        }
+
+        impl str::FromStr for $from {
+            type Err = FromHexError;
+
+            fn from_str(s: &str) -> Result<$from, FromHexError> {
+                let a = s.from_hex()?;
+                if a.len() != $size {
+                    return Err(FromHexError::InvalidHexLength);
+                }
+
+                let mut ret = [0;$size];
+                ret.copy_from_slice(&a);
+                Ok($from(ret))
+            }
+        }
+
+        impl fmt::Debug for $from {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                for i in &self.0[..] {
+                    write!(f, "{:02x}", i)?;
+                }
+                Ok(())
+            }
+        }
+
+        impl fmt::Display for $from {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                for i in &self.0[0..2] {
+                    write!(f, "{:02x}", i)?;
+                }
+                write!(f, "…")?;
+                for i in &self.0[$size - 2..$size] {
+                    write!(f, "{:02x}", i)?;
+                }
+                Ok(())
+            }
+        }
+
+        impl Copy for $from {}
+        #[cfg_attr(feature="dev", allow(expl_impl_clone_on_copy))]
```

Comment:
We don't have any dev features, should be removed and included or not included without dependency on it